__tests__/integration/config.test.ts

```diff
@@ -41,26 +44,48 @@ describe('config', () => {
     expect(await result.text()).toEqual('0\n1\n')
   })
 
-  it('should provide a custom logger implementation', async () => {
-    client = createTestClient({
-      log: {
-        enable: true,
-        LoggerClass: TestLogger,
-      },
+  describe('Logger support', () => {
+    const logLevelKey = 'CLICKHOUSE_LOG_LEVEL'
+    let defaultLogLevel: string | undefined
+    beforeEach(() => {
+      defaultLogLevel = process.env[logLevelKey]
+    })
+    afterEach(() => {
+      process.env[logLevelKey] = defaultLogLevel
     })
-    await client.ping()
-    expect(messages).toContainEqual(expect.stringContaining('GET /ping'))
-  })
 
-  it('should provide a custom logger implementation (but logs are disabled)', async () => {
-    client = createTestClient({
-      log: {
-        enable: false,
-        LoggerClass: TestLogger,
-      },
+    it('should provide a custom logger implementation', async () => {
+      process.env[logLevelKey] = 'DEBUG'
+      client = createTestClient({
+        log: {
+          // enable: true,
+          LoggerClass: TestLogger,
+        },
+      })
+      await client.ping()
+      console.log(logs)
+      // logs[0] are about current log level
+      expect(logs[1]).toEqual({
+        module: 'HTTP Adapter',
+        message: 'Got a response from ClickHouse',
+        args: expect.objectContaining({
+          request_path: '/ping',
+          request_method: 'GET',
+        }),
+      })
+    })
+
+    it('should provide a custom logger implementation (but logs are disabled)', async () => {
+      process.env[logLevelKey] = 'OFF'
```

Comment:
could you add a test for every log level please?
also let's verify the `console.log` renders an error message and points to an error stack trace